main.tf

```diff
@@ -139,9 +139,10 @@ module "bastion" {
   associate_public_ip_address = var.bastion_host_assign_public_ip
   enabled                     = var.bastion_host_enabled
   instance_type               = var.bastion_host_instance_type
-  key_name                    = aws_key_pair.bastion[0].key_name
+  key_name                    = var.bastion_host_ssh_key_name
```

Comment:
Hmm… I think this can't be done like this, otherwise https://github.com/isovalent/terraform-aws-vpc/blob/d50df2f2c9995eed770b3a33b3a3870d7c992c8d/main.tf#L125-L130 isn't considered anymore.
Someone might still want to not provide a custom key.

Reply:
Good point, but do you know why it's failing with:
`key_name                    = aws_key_pair.bastion[0].key_name`

```
│ Error: Reference to undeclared input variable
│ 
│   on .terraform/modules/vpc/main.tf line 142, in module "bastion":
│  142:   key_name                    = var.aws_key_pair.bastion[0].key_name
│ 
│ An input variable with the name "aws_key_pair" has not been declared. This variable can be declared with a variable "aws_key_pair" {} block.
```

Reply:
done